src/Sarif.Multitool/MultitoolOptionsBase.cs

```diff
@@ -26,10 +26,19 @@ public class MultitoolOptionsBase
             HelpText = "Output MD5, SHA1, and SHA-256 hash of analysis targets when emitting SARIF reports.")]
         public bool ComputeFileHashes { get; set; }
 
+        [Option(
+            't', // persist 't'ext
+            "persist-text-contents",
+            HelpText = "Persist a base64-encoded representation of all referenced textual files to the log.")]
```

Comment:
Again, UTF-8 transcoded.

Reply:
No sir. Not at this time.

---
In reply to: [200722202](https://github.com/Microsoft/sarif-sdk/pull/936#discussion_r200722202) [](ancestors = 200722202)

Reply:
Again, yes, per the spec.

Reply:
No, sire. The behavior of this command  is as follows: for any URI in the files table, if the extension of the file indicates it is a text file, base64 encode that file's contents into the file object Contents.Binary property. And so the help explanation here is accurate. In order to persist the file's contents to Contents.Text, we'd need to identify the encoding of the file on disk. There's no reliable mechanism to do this for all cases. Currently, our API doesn't allow us to do something like say 'assume any file you load from disk is UTF8'

---
In reply to: [201073313](https://github.com/Microsoft/sarif-sdk/pull/936#discussion_r201073313) [](ancestors = 201073313)

Reply:
Yup. As I said in the other comment I just sent (we crossed wires), I completely misunderstood the purpose of these options. And that means I completely misunderstood the product code as well, since I approved the PR. Sorry.